0000-spec-distribution.md

```diff
@@ -0,0 +1,307 @@
+# Meta 
+[meta]: #meta
+- Name: Buildpack Distribution Specification
+- Start Date: 2019-04-12
+- CNB Pull Requests: (spec PR to follow)
+- CNB Issues: (lifecycle issues to follow)
+
+
+# Motivation
+[motivation]: #motivation
+
+This proposal enables both decentralized, manual distribution of buildpacks from Docker registries as well as automatic distribution of buildpacks from a centralized buildpack registry.
+It allows individual buildpacks to be distributed via a Docker registry, and it makes dynamic assembly of builders on a Docker registry efficient.
+It provides a developer-friendly interface that abstracts away complex buildpack configurations.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes an official way to distribute buildpacks that conform to the CNB buildpack v3 specification.
+Changes would consist of a new Buildpack Distribution specification, modifications to the lifecycle builder, and modifications to the pack CLI.
+
+It affects all personas that interact with buildpacks.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Overview of Changes
+
+- The `order.toml` and `buildpack.toml` files are merged into `buildpack.toml`.
+- `buildpack.toml` can contain multiple buildpacks.
+- A buildpack inside of `buildpack.toml` is either:
+  a. a single buildpack implementation specified by a `path` field or
+  b. an ordering of other buildpack IDs/versions
+- Every buildpack inside `buildpack.toml` must have an ID and version.
+- Multiple buildpack implementations can be defined in `buildpack.toml`, so buildpacks can share files/code.
+- A buildpackage is a collection of buildpack repositories with `buildpack.toml` files.
+- A buildpackage can have a default buildpack.
+- Labels are removed from the specification.
+- The lifecycle moves into `/cnb/lifecycle/`.
+
+## Buildpack Blob Format
+
+A buildpack blob is a tgz containing a `buildpack.toml`. Example format:
+
+```toml
+[[buildpacks]]
+id = "io.buildpacks.nodejs"
+name = "Node.js Buildpack"
+version = "0.0.9"
+[[buildpacks.order]]
+group = [
+   { id = "io.buildpacks.node", version = "0.0.5" },
```

Comment:
can we make `version` optional if there's only 1 of that id specified?

Reply:
Only one of them where? A single version of the `order`-containing buildpack should represent an exact combination of versions of other buildpacks. Those buildpacks don't need to be defined in this file.